### PR TITLE
♻️ Preserve original file's ext in fetched temp file

### PR DESCRIPTION
**Pull request by znatty22**
Addresses #183

### CODE REVIEW

File: kf_lib_data_ingest/common/file_retriever.py [hunk 1]
```diff
@@ -182,18 +216,31 @@ def get(self, url, auth=None):
...
-                    protocol, path, self._files[url],
-                    auth=auth, logger=self.logger
-                )
+                # Settings for temporary file object that optionally deletes
+                # itself at the end of FileRetriever instance scope
+                # (if the enclosing folder isn't already being deleted)
+                delete = self.cleanup_at_exit and not self.__tmpdir
+
+                # Invoke http getter
+                # Note - Temp file must be created inside _web_save since we
```
Comment (znatty22) [inline kf_lib_data_ingest/common/file_retriever.py]: I tried to see if there was a way to set the NamedTemporaryFile's suffix after it's instantiated, but there doesn't seem to be a way unfortunately

File: kf_lib_data_ingest/common/file_retriever.py [hunk 2]
```diff
@@ -182,18 +216,31 @@ def get(self, url, auth=None):
...
+                # won't know file ext until after file is requested
+                if (protocol == 'http') or (protocol == 'https'):
```
Comment (znatty22) [inline kf_lib_data_ingest/common/file_retriever.py]: I tried to see if there was a way to set the NamedTemporaryFile's suffix after it's instantiated, but there doesn't seem to be a way unfortunately

**Comment by znatty22** [comment]
> Doing that wouldn't require any code outside of parsing the http response header.

Little confused by this. Regardless of what we do in terms of naming the temp file, we can't use the original source_data_url to get the file extension bc it may or may not have it. That's what extract stage was doing before: 
https://github.com/kids-first/kf-lib-data-ingest/blob/998f732048ce2c3816f3883e347cd7213d1c1ded/kf_lib_data_ingest/etl/extract/extract.py#L224
https://github.com/kids-first/kf-lib-data-ingest/blob/998f732048ce2c3816f3883e347cd7213d1c1ded/kf_lib_data_ingest/etl/extract/extract.py#L243

> Do we actually need to name the temp file, or is it sufficient to replace the name field or additionally set an original_name field on the returned data object and use that in extract?

I don't think we need to do anything differently? The tempfile coming from FileRetriever.get is assigned a random name (default behavior is random name has no file extension), and then the new code in this PR sets the extension to be the original file's extension using the `suffix` kwarg on NamedTemporaryFile

**Comment by fiendish** [comment]
I propose https://github.com/kids-first/kf-lib-data-ingest/pull/185

The modification to _web_save is more compliant with RFC 6266 and the modification to get is much simpler.

File: kf_lib_data_ingest/common/file_retriever.py [hunk 3]
```diff
@@ -88,23 +90,59 @@ def _web_save(protocol, source_loc, dest_obj, auth=None, logger=None):
...
     :param protocol: URL protocol identifier
     :type protocol: str
     :param source_loc: address or path
     :type source_loc: str
-    :param dest_obj: Receives the data downloaded from the source file
-    :type dest_obj: File-like object
+    :param dest_dir: Directory where temp file will be downloaded
+    :type dest_dir: str
+    :param delete: Flag indicating whether or not to delete file after download
+    :type delete: bool
```
Comment (fiendish) [inline kf_lib_data_ingest/common/file_retriever.py]: revert these param lines and the return line (though I'm not opposed to _also_ returning dest_obj. it just doesn't do that right now)

Comment (znatty22) [reply kf_lib_data_ingest/common/file_retriever.py]: None of the other getters return the dest obj so for now I'm going to revert back to what we had.

**Pull request merged**